command.go

```diff
@@ -233,6 +206,77 @@ func (c Command) Run(ctx *Context) (err error) {
 	return err
 }
 
+func getIndexes(ctx *Context) (int, int){
+	firstFlagIndex := -1
+	terminatorIndex := -1
+	for index, arg := range ctx.Args() {
+		if arg == "--" {
+			terminatorIndex = index
+			break
+		} else if arg == "-" {
+			// Do nothing. A dash alone is not really a flag.
+			continue
+		} else if strings.HasPrefix(arg, "-") && firstFlagIndex == -1 {
+			firstFlagIndex = index
+		}
+	}
+	if len(ctx.Args()) > 0 && !strings.HasPrefix(ctx.Args()[0], "-") && firstFlagIndex == -1{
+		return -1, -1
+	}
+
+	return firstFlagIndex, terminatorIndex
+
+}
+
+func copySlice(slice []string, start, end int) []string{
+	var newSlice []string
+	for i := start; i < end; i++ {
+		newSlice = append(newSlice, slice[i])
+	}
+	return newSlice
+}
+
+func getAllArgs(ctx *Context, firstFlagIndex, terminatorIndex int) (Args, []string) {
```

Comment:
It seems like this method only needs the args rather than the full `Context`. I think it could also benefit from some commenting, naming, or both of the return values. I think the return values should either both be `[]string` or `Args`.

Reply:
Fixed, let me know if not verbose enough